### PR TITLE
[doc] LLC quick start doc

### PR DESCRIPTION
**Pull request by msyyc**
https://github.com/Azure/azure-sdk-for-python/issues/21074

memo: another doc to guide how to write testcase for data-plane

### CODE REVIEW

**Comment by iscai-msft** [comment]
hey @msyyc sorry i can't get to it today, will look at your PR tomorrow. In the mean time have you taken a look at https://github.com/Azure/azure-sdk-for-python/issues/21074#issuecomment-935062698 as a first step?

File: scripts/dataplan_package/llc_initial.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    _LOGGER.info("Build start: %s", package_name)
+    check_parameters(output_folder)
+
+    # generate swagger readme
+    env = Environment(loader=FileSystemLoader(_TEMPLATE), keep_trailing_newline=True)
+    swagger_readme = generate_swagger_readme(output_folder, env, **kwargs)
+
+    # generate code with autorest and swagger readme
+    _LOGGER.info("generate SDK code with autorest")
+    check_call(f'autorest --version=3.4.5 --python --use=@autorest/python@5.8.4 --use=@autorest/modelerfour@4.19.2 --python-mode=update {swagger_readme}', shell=True)
```
Comment (lmazuel) [inline scripts/dataplan_package/llc_initial.py:80]: - I do not want to fix m4 in a cmdline, autorest should be responsible to do it
- we should us latest autorest.python
- we should make this tool generate with version-tolerant flags (talk to @iscai-msft )

**Comment by msyyc** [comment]
> hey @msyyc Yuchao Yan FTE sorry i can't get to it today, will look at your PR tomorrow. In the mean time have you taken a look at [#21074 (comment)](https://github.com/Azure/azure-sdk-for-python/issues/21074#issuecomment-935062698) as a first step?

Yes, but the difference is that I use the script instead of developing it in `autorest.python`. As I said, we could move the function to `autorest.python` later^_^

File: scripts/dataplan_package/llc_initial.py (hunk 1 above)
Comment (msyyc) [reply scripts/dataplan_package/llc_initial.py:80]: (1) To make package more stable, I think it's better to pin it like mgmt-plan package: https://github.com/Azure/azure-sdk-for-python/blob/6341dc07b2a81b16a403714ba34eaf17a2ae6ebb/swagger_to_sdk_config_autorest.json#L5.
(2) Of course, I will update the version before merge it.
(3) All the necessary configuration item is in swagger `README.md` so that customers needn't use complicated cmd:
image

File: scripts/dataplan_package/dev_requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+jinja2
```
Comment (lirenhe) [inline scripts/dataplan_package/dev_requirements.txt:1]: For the folder name, is it better to use dataplane_package?

File: scripts/dataplan_package/template/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+```python
+>>> from {{ namespace }} import {{ client_name }}
+>>> from azure.identity import DefaultAzureCredential
+>>> client = {{ client_name }}(endpoint='<endpoint>', credential=DefaultAzureCredential())
+```
+
+## Examples
+
+```python
+>>> from {{ namespace }} import {{ client_name }}
```
Comment (lirenhe) [inline scripts/dataplan_package/template/README.md:42]: why do we need '>>>' at the beginning?

File: scripts/dataplan_package/template/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+>>> from azure.identity import DefaultAzureCredential
+>>> from azure.core.exceptions import HttpResponseError
+
+>>> client = {{ client_name }}(endpoint='<endpoint>', credential=DefaultAzureCredential())
+>>> try:
+        <!-- wirte test code here -->
```
Comment (lirenhe) [inline scripts/dataplan_package/template/README.md:48]: typo

File: scripts/dataplan_package/llc_initial.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    _LOGGER.info("Build complete: %s", package_name)
+
+
+def main(**kwargs):
+    build_package(**kwargs)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="build package for Azure SDK of data-plan for Python",
```
Comment (lirenhe) [inline scripts/dataplan_package/llc_initial.py:114]: data-plane?

File: scripts/dataplan_package/llc_quick_start.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+
+# Prepare Environment(Windows)
+
+assume work path is `D:\`
+
+1. create virtual environment
+
+   ```
+   python venv venv-dev
```
Comment (lirenhe) [inline scripts/dataplan_package/llc_quick_start.md:17]: Is this command correct?
I use:  python -m venv venv-dev

File: scripts/dataplan_package/llc_quick_start.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+   .\venv-dev4\Scripts\Activate.ps1
```
Comment (lirenhe) [inline scripts/dataplan_package/llc_quick_start.md:18]: Should it be "venv-dev"?

File: scripts/dataplan_package/llc_quick_start.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+- `--client-name`: client name. For example: "WebPubSubServiceClient"
+- `--credential-scope`: each service for data plan has specific scopes. For example: " https://webpubsub.azure.com/.default"
+
+Here is the example, just try to run it:
+
+```
+python D:\azure-sdk-for-python\scripts\dataplan_package\llc_initial.py --output-folder D:\azure-sdk-for-python\sdk\llcexample\azure-messaging-webpubsubservic --input-file https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json --credential-scope https://webpubsub.azure.com/.default --package-name azure-messaging-webpubsubserviceexample --package-pprint-name "Azure WebPubSub Service" --client-name WebPubSubServiceClient
+```
+
+After running the scripts successfully, it creates necessary files and common test framework. **But remember to edit `README.md`, test files under `tests` and sample files under `samples`  since customers need more info to understand the package.**
```
Comment (lirenhe) [inline scripts/dataplan_package/llc_quick_start.md:63]: Could you provide more details of that?  Which file should I update for test and how can i run those tests even the empty test case successfully?

File: scripts/dataplan_package/llc_quick_start.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+You don't need it anymore after one-time successful run. Go to [Update](#Update) if you need to regenerate code.
+
+## Update
+
+After [Create](#Create), there will be `READMD.md` under `swagger` folder of `output_folder`. If need to update the code, run the following command:
+
+```
+autorest --version=3.6.2 --python --track2 --use=@autorest/python@5.11.1 --use=@autorest/modelerfour@4.19.2 --python-mode=update D:\azure-sdk-for-python\sdk\llcexample\azure-messaging-webpubsubserviceexample\swagger\README.md
```
Comment (lirenhe) [inline scripts/dataplan_package/llc_quick_start.md:72]: Not sure we should use the fixed version here.

File: scripts/dataplan_package/template/README.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+### Installating the package
+
+```bash
+python -m pip install {{ package_name }}
+```
+
+#### Prequisites
+
+- Python 2.7, or 3.6 or later is required to use this package.
+- You need an [Azure subscription][azure_sub] to use this package.
```
Comment (lirenhe) [inline scripts/dataplan_package/template/README.md:15]: Do we need to specify both [Azure subscription] and [azure_sub] here? I didn't see any replacement of those values.

File: scripts/dataplan_package/llc_quick_start.md (hunk 8 above)
Comment (msyyc) [reply scripts/dataplan_package/llc_quick_start.md:63]: I will write another doc to guide how to write testcase for data-plane

File: scripts/dataplan_package/llc_quick_start.md (hunk 9 above)
Comment (msyyc) [reply scripts/dataplan_package/llc_quick_start.md:72]: we fix the version to make sure customers will not block by possible unstable version. Once we think new version is stable enough, we could update the doc. It is also experience from mgmt-plane SDK release.

File: scripts/dataplan_package/template/README.md (hunk 10 above)
Comment (msyyc) [reply scripts/dataplan_package/template/README.md:15]: updated the doc link:
```
[azure_sub]: https://azure.microsoft.com/free/
```

File: scripts/dataplan_package/template/README.md (hunk 3 above)
Comment (msyyc) [reply scripts/dataplan_package/template/README.md:42]: It seems traditional for Python `readme.md` for data-plane.

File: scripts/dataplan_package/template/README.md (hunk 4 above)
Comment (msyyc) [reply scripts/dataplan_package/template/README.md:48]: It is obvious. I wan to tell that we need to add some case for `readme.md` after generation

File: scripts/dataplane_package/template/setup.py [hunk 11]
```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
```
Comment (iscai-msft) [inline scripts/dataplane_package/template/setup.py:2]: @msyyc can we take a step back and do this autorest issue first? https://github.com/Azure/azure-sdk-for-python/issues/21074#issuecomment-935062698

Sorry if this wasn't clear!

It's the one @lmazuel said is step 1. Sorry, should've made this more clear earlier.

As an aside, this part of the code looks copy-pasted from another script that does packaging. I'm not sure how @lmazuel wants this quickstart to look like in the end, but i think either way it would be better to call the other script from this script, instead of duplicating code

File: scripts/dataplan_package/dev_requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (msyyc) [reply scripts/dataplan_package/dev_requirements.txt]: Make sense. I think `quickstart_tooling_llc` is better folder name

File: scripts/dataplane_package/template/setup.py (hunk 11 above)
Comment (msyyc) [reply scripts/dataplane_package/template/setup.py:2]: Hi @iscai-msft, I am really working on the https://github.com/Azure/autorest.python/issues/1053. But after thinking, I think it is better to develop it with script first since we needn't rely on the new release of `autorest.python` for new function, which is more flexible. Once it is stable enough, we could move some function(like I [commented](https://github.com/Azure/autorest.python/issues/1053#issuecomment-964862928)) to `autorest.python` and simplify the quick-start doc.
As for the packaging, do you mean https://github.com/Azure/azure-sdk-for-python/tree/main/tools/azure-sdk-tools/packaging_tools/templates? It is for mgmt-plane as well. Maybe we could reuse it but I don't want to do that since there are many little difference. If we want to use other script, we need to change other script which is dangerous and high cost. I want to make the script more independent and stable.
@lirenhe tried with the tool and it could help to create basic package framework very quickly. Of course, we will continually optimize the quick start doc.

**Review by BigCat20196 — APPROVED** [review]

**Review by RAY-316 — APPROVED** [review]
